src/utils/permission/index.js

```diff
@@ -113,7 +114,7 @@ export const requestExternalStoragePermission = async () => {
         ? PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
         : PERMISSIONS.IOS.PHOTO_LIBRARY
     );
-    if (requestResult === RESULTS.GRANTED) {
+    if (requestResult === RESULTS.GRANTED || requestResult === RESULTS.LIMITED) {
       return {
         message: StringConstant.cameraPermissionGranted,
         success: true
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [117-121]

The change is good as it handles the `RESULTS.LIMITED` case, which is a new permission status in iOS 14 and later. This status allows the user to grant an app access to selected photos or files instead of the entire photo library or file system. However, please ensure that your application can handle this limited access correctly.
```diff
-    if (requestResult === RESULTS.GRANTED) {
+    if (requestResult === RESULTS.GRANTED || requestResult === RESULTS.LIMITED) {
       return {
         message: StringConstant.cameraPermissionGranted,
         success: true
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->